status/worksets.py

```diff
@@ -99,6 +99,29 @@ def get_workset_data(application, workset):
             for project in result[key]['projects']:
                 result[key]['projects'][project]['samples'] = dict(sorted(result[key]['projects'][project]['samples'].items()))
         return result
+        
+class ClosedWorksetsHandler(SafeHandler):
+    """ Handles all closed worksets for the closed ws tab
+    URL: /api/v1/closed_worksets
+    """
+    def get(self):
+        result = {}
+        a_year_ago = datetime.datetime.now() - relativedelta(years=1)
+        ws_view = self.application.worksets_db.view("worksets/summary", descending=True)
+        for row in ws_view:
+            try:
+                if parse(row.value['date_run']) <= a_year_ago:
+                    flag = True
+                    for project in row.value['projects']:
+                        if row.value['projects'][project]['status'] == 'Open':
+                            flag = False
+                            break
+                    if flag:
+                            result[row.key] = row.value
+            except TypeError:
```

Comment:
What? Why?

Reply:
Replaced the code with:

     `if row.value.get('date_run'):
                if parse(row.value['date_run']) <= a_year_ago:
                    flag = True
                    for project in row.value['projects']:
                        if row.value['projects'][project]['status'] == 'Open':
                            flag = False
                            break
                    if flag:
                            result[row.key] = row.value`

which works as well!